tests/components/nest/conftest.py

```diff
@@ -206,6 +225,8 @@ def config_entry(
             data[CONF_SUBSCRIBER_ID] = subscriber_id
         else:
             del data[CONF_SUBSCRIBER_ID]
+    if auth_implementation:
```

Comment:
I was easily able to remove the auth_implementation conditional, removing it from the static config to *always* be set here rather than an override.

That will be harder for subscriber id since it will explode the # of configs, but I will consider how to do this in a follow up (e.g. maybe one-off static configs for relevant tests that need to exercise this).  I am working on configuration.yaml deprecation which should remove some of the various states in the long term, which likely will need more focused tests exercising this in the short term.